velox/exec/tests/TopNRowNumberTest.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/tests/utils/AssertQueryBuilder.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::exec {
+
+namespace {
+
+class TopNRowNumberTest : public OperatorTestBase {};
+
+TEST_F(TopNRowNumberTest, basic) {
+  auto data = makeRowVector({
+      // Partitioning key.
+      makeFlatVector<int64_t>({1, 1, 2, 2, 1, 2, 1}),
+      // Sorting key
+      makeFlatVector<int64_t>({77, 66, 55, 44, 33, 22, 11}),
```

Comment:
It might be good to have some partition keys with the same order by keys for a variety in this data.

Reply:
@aditi-pandit Having duplicate order-by keys would make the query non-deterministic and therefore hard to verify results.